easypy/threadtree.py

```diff
@@ -258,3 +261,116 @@ def dump_threads():
     thread = concurrent(dump_threads, threadname="ThreadWatch", loop=True, sleep=interval)
     thread.start()
     _logger.info("threads watcher started")
+
+
+class ThreadContexts():
+    """
+    A structure for storing arbitrary data per thread.
+    Unlike threading.local, data stored here will be inherited into
+    child-threads (threads spawned from threads).
+
+        TC = ThreadContexts()
+
+        # (thread-a)
+        with TC(my_data='b'):
+            spawn_thread_b()
+
+        # (thread-b)
+        assert TC.my_data == 'b'
+        with TC(my_data='c'):
+            spawn_thread_c()
+        assert TC.my_data == 'b'
+
+        # (thread-c)
+        assert TC.my_data == 'c'
+
+    'accumulated': attributes named here get incremented each time, instead of overwritten:
```

Comment:
1. I could change them to `append` and `increment`
2. hmmm, we haven't needed that ourselves, so it doesn't seem like a common need. different apps might want to increase depth differently too